### PR TITLE
kube-aws: os-x compatible build script

### PR DESCRIPTION
**Pull request by colhom**

### CODE REVIEW

**Comment by colhom** [comment]
\cc @philips

**Comment by aknuds1** [comment]
Seems to work:
```
➜  aws git:(eec3fcb) ./build
Building kube-aws eec3fcbb6879e02888e41ec035233d0769962ea5...
# github.com/coreos/coreos-kubernetes/multi-node/aws/cmd/kube-aws
link: warning: option -X main.VERSION eec3fcbb6879e02888e41ec035233d0769962ea5 may not work in future releases; use -X main.VERSION=eec3fcbb6879e02888e41ec035233d0769962ea5
done
```

File: multi-node/aws/build [hunk 1]
```diff
@@ -15,10 +15,11 @@ if [ -n "$(git diff --shortstat 2> /dev/null | tail -n1)" ]; then
 fi
 
 echo "Building kube-aws ${VERSION}..."
-link_path="${PWD}/gopath/src/github.com/coreos/"
+rm -rf ./gopath
```
Comment (aaronlevy) [inline multi-node/aws/build]: Why is this necessary? The `ln -sf` should overwrite existing link.

File: multi-node/aws/build [hunk 2]
```diff
@@ -15,10 +15,11 @@ if [ -n "$(git diff --shortstat 2> /dev/null | tail -n1)" ]; then
...
+link_path="${PWD}/gopath/src/github.com/coreos"
 mkdir -p $link_path
 
-ln -sf "$(readlink -f ./../../)" "$link_path"
+ln -sf ${PWD}/../../ "${link_path}/coreos-kubernetes"
```
Comment (aaronlevy) [inline multi-node/aws/build]: nit: probably wouldn't hurt to clean up the link after the build.

File: multi-node/aws/build (hunk 1 above)
Comment (colhom) [inline multi-node/aws/build]: I just don't want that gopath dir hanging around. I want to discourage people from interacting with the gopath directory in general (erroneously go getting packages in there), so it's cleaned up after the build.

**Comment by colhom** [comment]
@aaronlevy let me know what you think.

File: multi-node/aws/build (hunk 1 above)
Comment (aaronlevy) [inline multi-node/aws/build]: gopath is a symlink, can you change this to `rm -f gopath`.

File: multi-node/aws/build [hunk 3]
```diff
@@ -15,10 +15,12 @@ if [ -n "$(git diff --shortstat 2> /dev/null | tail -n1)" ]; then
...
-link_path="${PWD}/gopath/src/github.com/coreos/"
+rm -rf ./gopath
+link_path="${PWD}/gopath/src/github.com/coreos"
 mkdir -p $link_path
-
-ln -sf "$(readlink -f ./../../)" "$link_path"
+ln -s ${PWD}/../../ "${link_path}/coreos-kubernetes"
 
 GOPATH=${PWD}/Godeps/_workspace:${PWD}/gopath CGO_ENABLED=0 go build -ldflags "-X main.VERSION ${VERSION}" -a -tags netgo -installsuffix netgo -o bin/kube-aws github.com/coreos/coreos-kubernetes/multi-node/aws/cmd/kube-aws
+rm -r ./gopath
```
Comment (aaronlevy) [inline multi-node/aws/build]: `rm ./gopath`

Comment (colhom) [inline multi-node/aws/build]: gopath is a directory. I would need to change the way the linking works...

Comment (aaronlevy) [inline multi-node/aws/build]: You're, right. I misread

**Comment by aaronlevy** [comment]
I guess if this works, lgtm...

**Comment by aaronlevy** [comment]
Nit: not even sure why this works (but it's late). FWIW, the etcd form of more or less the same thing: https://github.com/coreos/etcd/blob/master/build#L3-L10
